### ISSUE TITLE
Wrap `System.arraycopy`

### ISSUE THREAD

**Issue by mlutze** [flix/flix#6292]
For fast array manipulation. I think this could have a speedup for mutable data types that have to copy themselves occasionally.

**Comment by magnus-madsen** [comment]
I think we have `Array.copyOfRange`?

**Comment by mlutze** [comment]
That creates a copy that includes only the copied elements. I would like a function that copies into a larger array, e.g. for MutList expansion

**Comment by magnus-madsen** [comment]
I see

**Comment by magnus-madsen** [comment]
What should be the api?

**Comment by magnus-madsen** [comment]
Do we need type match or cast?

**Comment by mlutze** [comment]
Yes. The java interface actually uses `Object` instead of array for this function, so casting is needed, but no type match, from what I can tell.

**Comment by magnus-madsen** [comment]
Maybe we could just call it copyInto?

**Comment by JumpTwiice** [comment]
Shouldn't this be closed?
